doc/index.rst

```diff
@@ -0,0 +1,131 @@
+.. _index:
+
+.. image:: _static/images/npm_family_logo_primary.png
+   :align: center
+   :width: 200
+
+npmx-zephyr
+###########
+
+Overview
+********
+
+:ref:`npmx-zephyr <index>` provides an example of `npmx library`_ integration with the `Zephyr`_ RTOS environment.
+This can be used as a starting point for an integration with other RTOS or bare-metal environments.
+
+For the user's convenience, this repository includes also a variety of sample applications demonstrating how to use the `npmx drivers`_.
+Their code is available on the GitHub `npmx-zephyr repository`_.
+
+Supported devices
+*****************
+
+- nPM1300 Engineering B
+
+Supported drivers
+*****************
+The following matrix provides a comparative overview of which drivers are supported by specific Nordic nPMs.
+
+ ======================= ===== ====== ========= ======= ====== ====== ===== ===== ====== ======= ========
+         Driver           ADC   BUCK   CHARGER   EVENT   GPIO   LDSW   LED   POF   SHIP   TIMER   VBUSIN
+ ======================= ===== ====== ========= ======= ====== ====== ===== ===== ====== ======= ========
+  nPM1300 Engineering B    ✔     ✔        ✔        ✔      ✔      ✔      ✔     ✔     ✔       ✔       ✔
+ ======================= ===== ====== ========= ======= ====== ====== ===== ===== ====== ======= ========
+
+For more information about the features of a specific peripheral please refer to the official documentation of the specific nPM device.
+
+How to use
+**********
+
+Repository initialization
+=========================
+
+The recommended way to clone the repositories is by using the ``west`` tool.
+Information about the installation, available commands, configuration and more can be found in the `Zephyr's west`_ tool documentation.
+
+To initalize the npmx-zephyr repository, run the following commands:
+
+.. code:: bash
+
+   mkdir <new_directory>
+   cd <new_directory>
+   west init -m https://github.com/NordicSemiconductor/npmx-zephyr
+   cd npmx-zephyr
+   west update
+
+Alternatively, you can clone repositories by using git only.
+In this case pay special attention to revisions - both Zephyr and npmx revisions should match the ones specified in the ``west.yml``.
+Moreover the ``npmx`` folder should be placed in the ``zephyr/modules`` subdirectory.
+
+Building and running sample application
+=======================================
+
+Firstly, setup needs to be properly configured.
+Please read the documentation to learn how to do so.
+If you want to run one of the :doc:`sample applications <samples>`, follow the relevant instruction provided in the documentation for that specific sample.
+
+After configuring the setup, the application can be built and flashed to the target device.
+
+.. code:: bash
+
+   cd <path_to_npmx-zephyr>/samples/<desired_sample>
+   west build -b <target>
+   west flash
```

Comment:
There is no such thing as `irrelevant documentation`, I will go with:
 - "Configure the setup based on the documentation."

Reply:
@annakielar I've changed this one a little bit, because right know it doesn't make sense:

To run one of a :doc:`sample applications <samples>`, do the following:

* Configure the setup based on the relevant sample documentation.
* Build and flash the application to the target device by using the following commands:

  .. code:: bash

     cd <path_to_npmx-zephyr>/samples/<desired_sample>
     west build -b <target>
     west flash

For instance, if you want to run the LED sample for nPM1300 Engineering B used with nRF52840DK, you must configure the setup as specified in the :doc:`LED sample <../samples/led/README>` description.
After that, run the following commands:

.. code:: bash

   cd npmx-zephyr/samples/led
   west build -b nrf52840dk_nrf52840
   west flash

As a result, you should see LEDs blinking on the board.